design/cdk-bootstrap.md

```diff
@@ -0,0 +1,476 @@
+instead of granting those permissions to all principals in the account.
+
+The naming scheme will include the following elements in order to minimize the chance of name collisions:
+
+* The region we're bootstrapping in.
+* The account ID we're bootstrapping in.
+* The type of the resource (file assets bucket, Docker assets repository, etc.).
+
+### `--trust`
+
+We will add a new command-line option called `--trust` to the `bootstrap` command.
+Its value is a list of AWS account IDs:
+
+```shell
+$ cdk bootstrap [--trust accountId[,otherAccountId]...] aws:://account/region
+```
+
+If the `--trust` option has been passed on the CLI,
+the tool will present the user with the following "menu":
+
+```shell
+This will create a new IAM Role, trusting the CloudFormation service principal,
+that will be used for deploying your CloudFormation stacks in this environment.
+Please choose the Managed Policy or Policies to attach to the Role
+that contain the permissions required to perform those deployments:
+
+[1]  AdministratorAccess
+[2]  AmazonDynamoDBFullAccess
+[3]  AmazonEC2FullAccess
+[4]  AmazonECS_FullAccess
+[5]  AmazonS3FullAccess
+[6]  AmazonSageMakerFullAccess
+[7]  AmazonSESFullAccess
+[8]  AmazonSNSFullAccess
+[9]  AmazonSQSFullAccess
+[10] AWSLambdaFullAccess
+[11] SimpleWorkflowFullAccess
+...
+
+Please input the number of the chosen Policy from the above list,
+or provide a fully qualified ARN of a different Managed Policy.
+You can input multiple numbers and/or ARNs (up to 10),
+separated by commas.
+
+>
+```
+
+After the user has made their choice,
+or if the `--trust` option has not been passed,
+bootstrapping will proceed.
+
+We will also add a another option,
+`--cfn-deployment-role-policies`,
+that allows you to pass a list of managed policy ARNs on the command line,
+so that the `bootstrap` command can also be executed in non-interactive mode.
+
+### Removing existing customization options
+
+The existing customization options: `--bootstrap-bucket-name` and `--bootstrap-kms-key-id` will be removed.
+We will need to know the names of the bootstrap bucket and KMS key and synthesis time.
+The only way to customize the bootstrap template will be to deploy your own,
+based on the default one the CDK provides,
+and then change the default options when creating instances of the `Stack`
+class to match the names used in the custom template.
+
+### `--show-template`
+
+To help with the customization case,
+we will add a command that outputs the template we're using for bootstrapping to standard output.
+This way, it can serve as a base for anyone who wants customize the bootstrapping behavior.
+
+### CLI options in detail
+
+These options are inherited from the current CLI experience,
+and need to be kept for backwards compatibility reasons:
+
+* `--profile`: use the given local AWS credentials profile when interacting with the target environment.
+
+* `--toolkit-stack-name`: allows you to explicitly name the CloudFormation bootstrap stack
```

Comment:
Should be just called `cdk bootstrap --stack-name` (alias `--toolkit-stack-name` for backwards compat).